### PR TITLE
fix url logging

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

The `GetUri()` call contains side-effects which drain the builder of its constructed url data.  As a result, running a backup with CORSO_URL_LOGGING actually forces the bug to manifest unintentionally.

## Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included

## Type of change

- [x] :bug: Bugfix

## Test Plan

- [x] :muscle: Manual

### CODE REVIEW

File: src/internal/connector/exchange/api/mail.go [hunk 1]
```diff
@@ -286,16 +286,11 @@ func (c Mail) GetAddedAndRemovedItemIDs(
 	pgr := &mailPager{service, builder, options}
 
 	if len(os.Getenv("CORSO_URL_LOGGING")) > 0 {
-		gri, err := builder.ToGetRequestInformation(ctx, options)
+		gri, err := builder.ToGetRequestInformation(ctx, nil)
```
Comment (dadams39) [inline src/internal/connector/exchange/api/mail.go:289]: Is there a reason why the `options` are not being passed in?

File: src/internal/connector/exchange/api/events.go [hunk 2]
```diff
@@ -279,12 +279,7 @@ func (c Events) GetAddedAndRemovedItemIDs(
 		if err != nil {
 			logger.Ctx(ctx).Errorw("getting builder info", "error", err)
 		} else {
-			uri, err := gri.GetUri()
-			if err != nil {
-				logger.Ctx(ctx).Errorw("getting builder uri", "error", err)
-			} else {
-				logger.Ctx(ctx).Infow("calendar builder", "user", user, "directoryID", calendarID, "uri", uri)
-			}
+			logger.Ctx(ctx).Warnf("path-parameters %v", gri.PathParameters)
```
Comment (dadams39) [inline src/internal/connector/exchange/api/events.go:282]: Would it be helpfulto grab the gri.QueryParameters also and add them to the log?

File: CHANGELOG.md [hunk 3]
```diff
@@ -20,6 +20,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Errors from duplicate names in Exchange Calendars
 - Resolved an issue where progress bar displays could fail to exit, causing unbounded CPU consumption.
 - Fix Corso panic within Docker images
+- Debugging with the CORSO_URL_LOGGING env variable.
```
Comment (dadams39) [inline CHANGELOG.md:23]: This isn't clear

File: src/internal/connector/exchange/api/mail.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/connector/exchange/api/mail.go:289]: copy-paste error, thanks

File: src/internal/connector/exchange/api/events.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/internal/connector/exchange/api/events.go:282]: Not for the scope of the logging we want.

**Review by dadams39 — COMMENTED** [review]
@ryanfkeepers, going to go ahead and approve for when you get around to it. 
- The ChangeLog description --> CORSO_URL_LOGGING can be enabled for additional debugging
- The QueryParameter suggestion is to cater towards offering too much information with the hopes of being able to eliminate any other parameters that are causing the failure.

**Review by dadams39 — APPROVED** [review]

**Pull request merged**